### PR TITLE
Windows Updates

### PR DESCRIPTION
**Pull request by pderouen**
Fixes https://github.com/Origen-SDK/origen_testers/issues/175 and a handful of Windows related bugs.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
+#    `where origen`.split("\n").find do |match|
+#      match =~ /(.*)\\bin\\origen$/
+#    end
+#    origen_top = $1.gsub("\\", "/")
+#  else
+#    origen_top = `which origen`.strip.sub("/bin/origen", "")
+#  end
+#
+#  $LOAD_PATH.unshift "#{origen_top}/lib"
+# end
```
Comment (redxeth) [inline spec/spec_helper.rb:20]: So this works on windows now? If so maybe clear out the old comments?

Comment (pderouen) [reply spec/spec_helper.rb:20]: That code was checking to see if ../Gemfile exists. If it doesn't exist, it was checking the OS and which origen. None of that code was working correctly from Windows.

It all seems like it became outdated long ago (I don't know of a version of origen_testers that doesn't have Gemfile 1 directory up from spec/)

**Comment by pderouen** [comment]
@redxeth I removed those commented out lines of code from spec_helper

**Comment by redxeth** [comment]
What about the my other comment-- should more of pattern_compilers spec be under running_windows? check? Also I noticed a small indentation issue on https://github.com/Origen-SDK/origen_testers/blob/windows/spec/pattern_compilers_spec.rb#L608.   I guess lint doesn't run on the specs?

**Comment by pderouen** [comment]
> What about the my other comment-- should more of pattern_compilers spec be under running_windows? check? 

I only blocked the failing tests from running on Windows. For those origen_testers is generating an error if called while running on Windows.

> Also I noticed a small indentation issue on https://github.com/Origen-SDK/origen_testers/blob/windows/spec/pattern_compilers_spec.rb#L608. I guess lint doesn't run on the specs?

No, it doesn't run on specs. I noticed and fixed a few indentation issues when I was working on this PR. I'll see what I can do for it.

**Comment by pderouen** [comment]
@redxeth running lint on everything in the specs directory caused rspec to stop working. So, I ran lint just on the 2 files I touched.

**Review by redxeth — APPROVED** [review]

File: spec/pattern_compilers_spec.rb [hunk 2]
```diff
@@ -550,60 +550,62 @@ def configs
...
-        dut.v93k_compiler_options[:clean] = true
-        dut.add_pattern_compiler(:id1, :v93k, dut.v93k_compiler_options)
-        dut.pattern_compilers[:id1].run
-      end
-
-      it 'runs existing aiv file' do
-        dut.pattern_compilers.should == {}
-        dut.v93k_compiler_options[:clean] = true
-        dut.add_pattern_compiler(:id1, :v93k, dut.v93k_compiler_options)
+      unless Origen.running_on_windows?
```
[the remark this answers is not in the document]
Comment (pderouen) [reply spec/pattern_compilers_spec.rb:543]: I'm sort of confused as to why pattern compilation is designed to error out when running on Windows. It makes sense for 93k I suppose. But, the compilers for Teradyne patterns are native to Windows. At some point we might want to change the code to check if the compile command is available on the system. Though I don't think there's a way to have them installed on the CI Windows machine. That can be an update for a future PR (or not).

**Pull request merged**